src/data/mat.c

```diff
@@ -271,3 +271,105 @@ void cfe_mat_mul(cfe_mat *res, cfe_mat *m1, cfe_mat *m2) {
     }
     mpz_clears(sum, prod, x, y, NULL);
 }
+
+void cfe_mat_extract_submatrix(cfe_mat *m, cfe_mat *min, size_t i, size_t j) {
+    assert(i < m->rows && j < m->cols);
```

Comment:
Can you add an additional assert to verify that the dimensions of `min` are `rows-1 x cols-1`?

Reply:
With this comment I remembered that I could also add dimension check in derive key and encrypt.

Reply:
Great!